### PR TITLE
Dt/oc 2550/client rename

### PR DESCRIPTION
**Pull request by doubt72**
Rename plus error handling for clients

### CODE REVIEW

File: src/chef_wm_named_client.erl [hunk 1]
```diff
@@ -96,21 +97,34 @@ from_json(Req, #base_state{reqid = RequestId,
...
+                           Req1;
+                       _ ->
+                           chef_wm_util:append_field_to_json_body(Req1,
+                                                                  <<"private_key">>,
+                                                                  PrivateKey)
+                   end,
+            % Need to return uri (in case request is a rename?)
+            Uri = ?BASE_ROUTES:route(client, Req1, [{name, Name}]),
+            FinalReq = chef_wm_util:append_field_to_json_body(Req2, <<"uri">>, Uri),
+            case ReqName of
```
Comment (seth) [inline src/chef_wm_named_client.erl]: This part could use a comment or maybe even a small helper function. The check is whether this was a regular update or a rename and we should make that clear.

Comment (seth) [inline src/chef_wm_named_client.erl]: Although I see how you named the vars, so nice touch on that :)

**Comment by seth** [comment]
Might make sense to add to this PR an update that pulls in the configurable mechanism for generating keys. Look at the chef_wm_clients module and commit 7df54d751780bcec11fe9ea4b8fa4d.

**Comment by seth** [comment]
Overall looks very close. There are some other bits that could use a bit of refactoring but not part of these diffs so we'll pick them up later.

File: src/chef_wm_named_client.erl (hunk 1 above)
Comment (doubt72) [inline src/chef_wm_named_client.erl]: Well, as clever or not naming the var was, I think you're right and a helper function seemed a bit cleaner.

**Comment by seth** [comment]
merge when ready

**Pull request merged**